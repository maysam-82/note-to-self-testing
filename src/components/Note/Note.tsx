import React from 'react';

interface INote {
	note: string;
	id: number;
}

const Note: React.SFC<INote> = ({ note, id }) => {
	return <div className="mt-1 border rounded p-2 text-left">{note}</div>;
};

export default Note;
