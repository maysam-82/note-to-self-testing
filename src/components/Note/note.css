.note {
	width: 90%;
}
.delete {
	width: 10%;
	cursor: pointer;
}
