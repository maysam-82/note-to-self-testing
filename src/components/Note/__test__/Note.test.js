import React from 'react';
import { shallow } from 'enzyme';
import Note from '../Note';

describe('Note Component', () => {
	const mockDelete = jest.fn();
	const mockEdit = jest.fn();
	const props = {
		note: 'test note',
		id: 100,
		handleDeleteNote: mockDelete,
		handleEditNote: mockEdit,
	};
	const component = shallow(<Note {...props} />);
	it('should render Note component', () => {
		expect(component).toMatchSnapshot();
	});
	it('should render `test note`', () => {
		expect(component.find('.note').text()).toEqual(props.note);
	});
	describe('when click on delete icon', () => {
		component.find('.delete').simulate('click');
		it('should call handleDeleteNote', () => {
			expect(mockDelete).toHaveBeenCalledWith(props.id);
		});
	});
	describe('when click on edit icon', () => {
		component.find('.edit').simulate('click');
		it('should call handleEditNote', () => {
			expect(mockEdit).toHaveBeenCalledWith(props.id);
		});
	});
});
