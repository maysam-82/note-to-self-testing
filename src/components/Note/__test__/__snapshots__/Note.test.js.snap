// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note Component should render Note component 1`] = `
<div
  className="d-flex border rounded align-items-center mt-1"
>
  <div
    className="  p-2 text-left note"
  >
    test note
  </div>
  <div
    className="edit pt-2 pb-2"
    onClick={[Function]}
  >
    ✍
  </div>
  <div
    className="delete pt-2 pb-2"
    onClick={[Function]}
  >
    ✖
  </div>
</div>
`;
